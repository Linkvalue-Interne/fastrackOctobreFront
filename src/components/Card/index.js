import React from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

import CardPartner from '../commons/cardPartner';
import PictureBox from '../commons/picture';
import SpanBold from '../commons/spanBold';
import H3 from '../commons/titleH3';
import P from '../commons/paragraphe';
import Img from '../commons/logoTechno';

const Container = styled.div`
  display: flex;
  width: calc(100% - 80px);
  height: 95vh;
  margin-left: 80px;
  padding: 7rem;
  position: relative;
  flex-wrap: wrap;
`;

const CustomH3 = styled(H3)`
  padding: 1rem;
`;

const LogoContainer = styled.div`
  display: flex;
  padding: 1.5rem 3rem;;
  width: 100%;
  justify-content: center;
  align-items: center;
`;

const Logo = styled.div`
  display: flex;
  width: calc(100% / 3);
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

const Card = (props) => (
  <Container>
    <CardPartner>
      <PictureBox image={props.image} />
      <CustomH3>{props.firstname} <SpanBold>{props.lastname}</SpanBold></CustomH3>
      <P>{props.jobs}</P>
      <LogoContainer>
        <Logo>
          <Img src={props.firstTechno} alt=""/>
          <SpanBold><P>{props.firstTechnoName}</P></SpanBold>
        </Logo>
        <Logo>
          <Img src={props.secondTechno} alt=""/>
          <SpanBold><P>{props.secondTechnoName}</P></SpanBold>
        </Logo>
        <Logo>
          <Img src={props.thirdTechno} alt=""/>
          <SpanBold><P>{props.thirdTechnoName}</P></SpanBold>
        </Logo>
      </LogoContainer>
    </CardPartner>
  </Container>
);

Card.propTypes = {
  image: PropTypes.string.isRequired,
  firstname: PropTypes.string.isRequired,
  lastname: PropTypes.string.isRequired,
  jobs: PropTypes.string.isRequired,
  firstTechno: PropTypes.string.isRequired,
  secondTechno: PropTypes.string.isRequired,
  thirdTechno: PropTypes.string.isRequired,
  firstTechnoName: PropTypes.string.isRequired,
  secondTechnoName: PropTypes.string.isRequired,
  thirdTechnoName: PropTypes.string.isRequired,

};

export default Card;
