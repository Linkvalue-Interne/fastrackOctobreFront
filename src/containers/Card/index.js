import React from 'react';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';

import CardPartner from '../../commons/cardPartner';
import PictureBox from '../../commons/picture';
import Img from '../../commons/logoTechno';
import { H3, P, SpanBold } from '../../commons/text';

const CustomH3 = styled(H3)`
  padding: 1rem;
`;

const LogoContainer = styled.div`
  display: flex;
  padding: 1.5rem 3rem;;
  width: 100%;
  justify-content: center;
  align-items: center;
`;

const Logo = styled.div`
  display: flex;
  width: calc(100% / 3);
  justify-content: center;
  align-items: center;
  flex-direction: column;
`;

const Card = ({ partner }) => (
  <CardPartner>
    <PictureBox image={partner.image} />
    <CustomH3>{partner.firstname} <SpanBold>{partner.lastname}</SpanBold></CustomH3>
    <P>{partner.jobs}</P>
    <LogoContainer>
      <Logo>
        <Img src={partner.firstTechno} alt="" />
        <SpanBold><P>{partner.firstTechnoName}</P></SpanBold>
      </Logo>
      <Logo>
        <Img src={partner.secondTechno} alt="" />
        <SpanBold><P>{partner.secondTechnoName}</P></SpanBold>
      </Logo>
      <Logo>
        <Img src={partner.thirdTechno} alt="" />
        <SpanBold><P>{partner.thirdTechnoName}</P></SpanBold>
      </Logo>
    </LogoContainer>
  </CardPartner>
);

Card.propTypes = {

  partner: PropTypes.shape({
    image: PropTypes.string,
    firstname: PropTypes.string.isRequired,
    lastname: PropTypes.string.isRequired,
    jobs: PropTypes.string.isRequired,
    firstTechno: PropTypes.string.isRequired,
    secondTechno: PropTypes.string,
    thirdTechno: PropTypes.string,
    firstTechnoName: PropTypes.string.isRequired,
    secondTechnoName: PropTypes.string,
    thirdTechnoName: PropTypes.string,
  }).isRequired,
};

export default Card;
