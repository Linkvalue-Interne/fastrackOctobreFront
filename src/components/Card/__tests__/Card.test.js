import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import Card from '../index';

configure({ adapter: new Adapter() });

describe('Card', () => {
  it('should render without crashing', () => {
    const render = shallow(<Card
        
      />);
    expect(render).toMatchSnapshot();
  });
});
