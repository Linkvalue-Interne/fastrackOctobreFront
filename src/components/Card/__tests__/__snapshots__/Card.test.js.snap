// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render without crashing 1`] = `
<Delayed>
  <Styled(div) />
  <Styled(h3)>
     
    <Styled(span) />
  </Styled(h3)>
  <Styled(p) />
  <Styled(div)>
    <Styled(div)>
      <Styled(img)
        alt=""
      />
      <Styled(span)>
        <Styled(p) />
      </Styled(span)>
    </Styled(div)>
    <Styled(div)>
      <Styled(img)
        alt=""
      />
      <Styled(span)>
        <Styled(p) />
      </Styled(span)>
    </Styled(div)>
    <Styled(div)>
      <Styled(img)
        alt=""
      />
      <Styled(span)>
        <Styled(p) />
      </Styled(span)>
    </Styled(div)>
  </Styled(div)>
</Delayed>
`;
